types/Impl/Product.php

```diff
@@ -0,0 +1,40 @@
+<?php
+declare(strict_types=1);
+
+namespace Garp\Functional\Types\Impl;
+
+use Garp\Functional\Types\{Semigroup, Setoid};
+
+/**
+ * @package  Garp\Functional
+ * @author   Harmen Janssen <harmen@grrr.nl>
+ * @license  https://github.com/grrr-amsterdam/garp-functional/blob/master/LICENSE.md BSD-3-Clause
+ * @see      https://github.com/fantasyland/fantasy-land#semigroup
+ */
+class Product implements Semigroup, Setoid {
```

Comment:
Might be good to make `Type\Impl` classes `final` to force using the type interfaces.

Reply:
Yes, good idea, will do. 👍